README.md

```diff
@@ -48,16 +48,18 @@ Please, read the [getting started](https://doc.bblf.sh/using-babelfish/getting-s
 import bblfsh
 
 client = bblfsh.BblfshClient("0.0.0.0:9432")
-uast = client.parse("/path/to/file.py").uast
-print(uast)
+uast = client.parse("/path/to/file.py")
+print(uast.load())
+
 # "filter' allows you to use XPath queries to filter on result nodes:
-print(bblfsh.filter(uast, "//Import[@roleImport and @roleDeclaration]//alias"))
+it = uast.filter("//Import[@role='Import' and @role='Declaration']//alias")
+for node in it:
+    print(node.load())
 
-# filter\_[bool|string|number] must be used when using XPath functions returning
-# these types:
-print(bblfsh.filter_bool(uast, "boolean(//*[@strtOffset or @endOffset])"))
-print(bblfsh.filter_string(uast, "name(//*[1])"))
-print(bblfsh.filter_number(uast, "count(//*)"))
+# filter must be used when using XPath functions returning these types:
```

Comment:
The comment about the types is spurious if filter now works with different types.

Reply:
One worry I have with this version of the client is that it seems to be backward incompatible with existing code (as these functions and the returned value show) which is not something I'm aware was advertised since the rewrite of libuast should have keep the clients APIs the same (or at least I'm pretty sure most users of the Python client are not aware of this).

Reply:
But it will be incompatible in any case, since:
- `Node` does not exist and I don't think we should emulate it.
- `load()` was not used in the old code, but we need it for better performance.
- `filter` returned a node or an array, but now it will return an iterator.

We can still emulate `filter_*` if necessary. But I think a generic `filter_one` or `filter_val` will be better, since Python is not strictly typed, and it's nice to enter the query and get a value back whatever the type is.

Reply:
Tagging @vmarkovtsev to share opinions about backward compatibility.

Reply:
filter() returned a Python iterator already, not arrays or single nodes. I'm not saying we should not improve, just that we should not break backwards compatibility without checking for impact before with users and management.

Reply:
On the leads meeting @mcuadros didn't saw any problem breaking backward compatibility on v3 so if @vmarkovtsev doesn't have a problem with that either I think we can go ahead with this.

Reply:
Having solved the compatibility stuff and returning to the other part, I still think that a single `filter` function returning different types is worse than a function named for every type (because it's super unidiomatic and forces the user to check the type on every call and doesn't check if the user is getting the wrong/unexpected type).

Reply:
It relies on the returned type of the query here. `name(*)` always returns a string, for example.

Reply:
We don't use those removed functions

Reply:
Because you didn't use queries returning anything outside of nodes, but that doesn't mean they should be removed (for example the sonar checks used some of them IIRC).